File: Task_2/task_2.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution{
    public:
        int findKthLargest(vector<int>& nums, int k){
            vector<int> minstack(k);
            for(int i = 0; i < k; i++){
                minstack[i] = nums[i];
            }
            // 创建小顶堆
            make_heap(minstack.begin(), minstack.end(), greater<int>()); 
            for(int i = k; i < nums.size(); i++){
                // 如果比根节点大 插入堆中
                if(nums[i] > minstack[0]){
                    minstack.push_back(nums[i]);
                    push_heap(minstack.begin(), minstack.end(), greater<int>());
                }
                // 保持堆的大小为k
                if(minstack.size() > k){
                    pop_heap(minstack.begin(), minstack.end(), greater<int>());
                    minstack.pop_back();
                }
            }
            // 堆顶元素即为第k个最大的元素
            return minstack[0];
        }
};

File: Task_2/task_2.py
import heapq
class Solution:
	def findKthLargest(self, nums, k):
		heap = []
		for n in nums[:k]:
			heapq.heappush(heap, n)
		for n in nums[k:]:
			if n > heap[0]:
				heapq.heappop(heap)
				heapq.heappush(heap, n)
		return heap[0]


if __name__ == '__main__':
	s = Solution()
	nums = [3,2,1,5,6,4]
	k = 2
	result = s.findKthLargest(nums, k)
	print(result)